File: source/android/rooOpenUDID.java
/*
java implementation of the rooOpenUDID extension.

Add android-specific functionality here.

These functions are called via JNI from native code.
*/
/*
 * NOTE: This file was originally written by the extension builder, but will not
 * be overwritten (unless --force is specified) and is intended to be modified.
 */
import com.ideaworks3d.marmalade.LoaderAPI;
import com.ideaworks3d.marmalade.LoaderActivity;


class rooOpenUDID
{
    public void rooOpenUDID_init()
    {
         org.OpenUDID.OpenUDID_manager.sync(LoaderActivity.m_Activity.getApplicationContext());
    }
    public boolean rooOpenUDID_isInitialized()
    {
        return org.OpenUDID.OpenUDID_manager.isInitialized();
    }
    public String rooOpenUDID_getValue()
    {
        return org.OpenUDID.OpenUDID_manager.getOpenUDID();
    }
}

File: source/generic/rooOpenUDID.cpp
/*
Generic implementation of the rooOpenUDID extension.
This file should perform any platform-indepedentent functionality
(e.g. error checking) before calling platform-dependent implementations.
*/

/*
 * NOTE: This file was originally written by the extension builder, but will not
 * be overwritten (unless --force is specified) and is intended to be modified.
 */


#include "rooOpenUDID_internal.h"
s3eResult rooOpenUDIDInit()
{
    //Add any generic initialisation code here
    return rooOpenUDIDInit_platform();
}

void rooOpenUDIDTerminate()
{
    //Add any generic termination code here
    rooOpenUDIDTerminate_platform();
}

void rooOpenUDID_init()
{
	rooOpenUDID_init_platform();
}

bool rooOpenUDID_isInitialized()
{
	return rooOpenUDID_isInitialized_platform();
}

const char * rooOpenUDID_getValue()
{
	return rooOpenUDID_getValue_platform();
}
